mailbox/api/src/main/java/org/apache/james/mailbox/searchhighligt/SearchHighlighter.java

```diff
@@ -0,0 +1,29 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.searchhighligt;
+
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.model.MultimailboxesSearchQuery;
+import org.reactivestreams.Publisher;
+
+public interface SearchHighlighter {
+
+    Publisher<SearchSnippet> highlightSearch(MultimailboxesSearchQuery expression, MailboxSession session, long limit);
```

Comment:
![image](https://github.com/user-attachments/assets/5a1e7ace-1131-4369-b3e3-966709d407cb)

```suggestion
    Publisher<SearchSnippet> highlightSearch(List<MessageId> messageIds, MultimailboxesSearchQuery expression, MailboxSession session);
```

To closer stick to JMAP, IMO

Reply:
`List<MessageId> messageIds` already in `MultimailboxesSearchQuery expression`
`long limit` is necessary for search

Reply:
When search high light, we have 2 step
- 1) search query 
- 2) search hight light base on (1) result.
I feel your mind is `highlightSearch` is only (2)
But as my undstand `highlightSearch` is combine (1) and (2)

Please correct me if wrong

Reply:
limit = messageIds.size isn't it?

Reply:
ah, messageIds, not mailboxid